### DISCUSSION TITLE
Possibility to change path for assets at runtime, without changing router base url (SSR)

### DISCUSSION THREAD

**Discussion by sebsauer90** [quasarframework/quasar#13415] [Ideas / Proposals]
First thanks for this awesome framework :)

We need to change the path for assets at runtime, but without changing the router base url.
Assets like js and css should be available here: example-domain.de/asset-path/
But the apllication path should still be: example-domain.de

If we use the quasar config `publicPath: 'asset-path'` the whole application is running on /asset-path/.

Nuxt does it here: [https://nuxtjs.org/docs/configuration-glossary/configuration-build/#publicpath](https://nuxtjs.org/docs/configuration-glossary/configuration-build/#publicpath)

So, is there a way to configure this behaviour in quasar?
I think it would also be very useful to publish assets via a CDN.

Thanks a lot!

**Comment by pdanpdan** [comment]
In your specific case just put your assets in the asset-path subfolder of public

**Reply by sebsauer90** [reply]
Hey [pdanpdan](https://github.com/pdanpdan), this dosen't work for the builded js and css files by quasar. We need all files on "asset-path", also the quasar generated files.

**Comment by pdanpdan** [comment]
I think you need to play with https://quasar.dev/quasar-cli-webpack/quasar-config-js#property-build (search for `vueRouterBase`)
Then you would set the `publicPath` to asset-path and `vueRouterBase` to '/' I think

**Reply by sebsauer90** [reply]
That was my idea too and i tried out each possible configuration, but without a result :(

**Comment by pdanpdan** [comment]
Q1 or Q2? vite or webpack?

**Reply by sebsauer90** [reply]
Q2 with webpack. Thanks in advance for your time!

**Comment by pdanpdan** [comment]
I just tested:

quasar.conf.js
```
build: {
  vueRouterMode: 'history', // available values: 'hash', 'history'
  vueRouterBase: '/',
  publicPath: '/assets',
  ...
}
```

```bash
quasar build
cd dist/spa
mkdir assets
mv css/ fonts/ icons/ js/ assets
quasar serve . -p 7070 -s -c 0 --history
```

All works as expected

**Reply by sebsauer90** [reply]
Hey @pdanpdan, this dosen't work for SSR.
And the whole application is running under the url mydomain.com/_assets_sf because of the publicPath config
